### PR TITLE
fetch all tasks from backend before sleep

### PR DESCRIPTION
**Pull request by elfkuzco**
# Changes
- fetch all pending tasks from worker before sleeping using generators

### CODE REVIEW

File: worker/manager/src/mirrors_qa_manager/worker.py [hunk 1]
```diff
@@ -291,16 +292,29 @@ def update_countries_list(self):
...
+            current_page = data["metadata"]["current_page"]
+            last_page = data["metadata"]["last_page"]
+
+            logger.debug(
+                f"Fetched page {current_page} of {last_page} of pending tests."
+            )
 
-        logger.info(f"Fetched {data['metadata']['page_size']} test(s) from Backend API")
+            if current_page == last_page:
+                raise StopIteration
```
Comment (rgaudin) [inline worker/manager/src/mirrors_qa_manager/worker.py:315]: Wont this prevent the yield stmt to be called for the last page?

**Review by rgaudin — CHANGES_REQUESTED** [review]
Please look at the question

File: worker/manager/src/mirrors_qa_manager/worker.py (hunk 1 above)
Comment (elfkuzco) [reply worker/manager/src/mirrors_qa_manager/worker.py:315]: Oh, wow. Thanks for pointing that out.

**Review by rgaudin — APPROVED** [review]
👍

**Pull request merged**